src/storage-adapters/mongo/index.test.js

```diff
@@ -0,0 +1,501 @@
+      });
+
+      it('stores the document type', () => {
+        expect(mongoRecord.documentType).to.include(DOCUMENT_TYPE);
+      });
+
+      it('stores information that it is the first record for this specific document', () => {
+        expect(mongoRecord.isFirstRecord).to.be.true;
+      });
+
+      it('stores the proper content', () => {
+        expect(mongoRecord.content).to.equal(CONTENT);
+      });
+
+      context('when provided', () => {
+        it('stores the fetch date', () => {
+          expect(new Date(mongoRecord.fetchDate).getTime()).to.equal(FETCH_DATE.getTime());
+        });
+
+        it('stores the mime type', () => {
+          expect(mongoRecord.mimeType).to.equal(MIME_TYPE);
+        });
+
+        it('stores the snapshot ID', () => {
+          expect(mongoRecord.snapshotId.toString()).to.include(SNAPSHOT_ID);
+        });
+      });
+    });
+
+    context('when it is not the first record', () => {
+      const UPDATED_CONTENT = `${CONTENT} updated`;
+
+      before(async () => {
+        (record = await subject.record({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+          content: CONTENT,
+          mimeType: MIME_TYPE,
+          fetchDate: FETCH_DATE,
+          snapshotId: SNAPSHOT_ID,
+        }));
+
+        numberOfRecordsBefore = await collection.find({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+        }).count();
+
+        (record = await subject.record({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+          content: UPDATED_CONTENT,
+          mimeType: MIME_TYPE,
+          fetchDate: FETCH_DATE,
+          snapshotId: SNAPSHOT_ID,
+        }));
+
+        numberOfRecordsAfter = await collection.find({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+        }).count();
+
+        ([mongoRecord] = await collection.find({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+        }).limit(1).sort({ created_at: -1 }).toArray());
+      });
+
+      after(() => collection.drop());
+
+      it('saves the record', () => {
+        expect(numberOfRecordsAfter).to.equal(numberOfRecordsBefore + 1);
+      });
+
+      it('returns the record id', () => {
+        expect(mongoRecord._id.toString()).to.equal(record.id);
+      });
+
+      it('returns a boolean to know if it is the first record', () => {
+        expect(record.isFirstRecord).to.be.false;
+      });
+
+      it('stores the service id', () => {
+        expect(mongoRecord.serviceId).to.include(SERVICE_PROVIDER_ID);
+      });
+
+      it('stores the document type', () => {
+        expect(mongoRecord.documentType).to.include(DOCUMENT_TYPE);
+      });
+
+      it('stores information that it is the first record for this specific document', () => {
+        expect(mongoRecord.isFirstRecord).to.be.undefined;
```

Comment:
Because `expect` library does not provide something like `expect(mongoRecord.isFirstRecord).to.be.falsy` the only alternative here is `expect(mongoRecord.isFirstRecord).to.not.be.ok`, which I found confusing.

Reply:
I was not so concerned about checking `undefined` instead of falsy, but more about checking that the Mongo object has a `isFirstRecord` property set to some specific value. This is whitebox testing, relying on the implementation of the database storage layer. The API that I am concerned with checking is that the `isFirstRecord` property of the Record object is `false`, not how it is implemented 🙂

Reply:
Maybe I'm wrong but I think you are confused between the MongoAdapter API and the Recorder API. Let's talk about this.